### ISSUE TITLE
Log file for refine_viaf ?

### ISSUE THREAD

**Issue by JBPressac** [codeforkjeff/refine_viaf#2]
Hello,
Thank you for this very usefull tool. I tried to used it to reconcile a list of authors writing in Breton with BnF FRBNF but the _best candidate's score_ facet indicates 741 errors for 1178 rows. I noticed that the Windows console from which I lauched refine_viaf-1.4.jar displayed errors like 

`Exception in thread "pool-452-thread-2" java.lang.NullPointerException`

or

```
Exception in thread "pool-434-thread-3" java.lang.NullPointerException
        at com.codefork.refine.viaf.sources.NonVIAFSource.formatResult(NonVIAFSource.java:92)
        at com.codefork.refine.viaf.VIAF.doSearch(VIAF.java:230)
        at com.codefork.refine.viaf.VIAF.search(VIAF.java:170)
        at com.codefork.refine.SearchThread.run(SearchThread.java:23)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
```

or

```
2016-06-15 14:26:06.215 ERROR 3060 --- [ol-395-thread-1] com.codefork.refine.viaf.VIAF            : ioerror: java.io.IOException: Server returned HTTP response code: 429 for URL: http://www.viaf.org/viaf/search?query=local.personalNames%20all%20%22emmanuel%20talec%22%20and%20local.sources%20%3D%20%2
2bnf%22&sortKeys=holdingscount&maximumRecords=9&httpAccept=application/xml
```

But is there a way to have a log file of those errors to analyse them and to see which request generated an error.

Thanks,

**Comment by codeforkjeff** [comment]
Hi,

Thank you for reporting these problems!

I created a new release v1.4.1 to fix the NullPointerException errors. Please let me know if it resolves them.

If you still see those errors, please try running it again using http://refine.codefork.com/ so that I can see the errors in my own logs and track them down better. 

The last error "Server returned HTTP response code: 429" means that you are making too many requests to VIAF and they are rate-limiting your queries. This is a "normal" error that can happen when you have multiple OpenRefine applications using the same service at once. This error will cause that name (in this case "emmanuel talec") to have zero candidates in OpenRefine.

I'll also work on improving the log error messages in general,  but that will have to wait a few days.

**Comment by JBPressac** [comment]
Hello,
Thank you for the update, I will try it.

 You said:

> The last error "Server returned HTTP response code: 429" means that you are making too many requests to VIAF and they are rate-limiting your queries. This is a "normal" error that can happen when you have multiple OpenRefine applications using the same service at once.

However, I lauched only one reconciliation at once from OpenRefine (sorry, I did not mention that I am using refine_viaf on my personal computer running Windows).

I will lauch the reconciliation from your server so you could see the errors.

**Comment by codeforkjeff** [comment]
That is very strange. I don't know why you are getting those 429 responses, then. I've only seen them once before. If they continue, you might want to contact VIAF directly and ask if they put special restrictions on your network and/or IP address.

I made release v1.4.2. When an error occurs, you should now see the name that failed in the log file.

**Comment by JBPressac** [comment]
Hello,
I lauched the reconciliation of my 1178 rows dataset on http://refine.codefork.com/reconcile/viafproxy/BNF at 7:00 UTC (9:00 in France). The score was better than my local version of refine_viaf-1.4.jar as I "only" had 616 errors (versus 741).

As my data contains accents (ê, à, etc.) and other french characters (like ç) I removed the accents with Jython on my requests:

```
import unicodedata;
return unicodedata.normalize('NFKD', value).encode('ascii', 'ignore')
```

then I made a second reconciliation on http://refine.codefork.com/reconcile/viafproxy/BNF at 7:13 UTC (9:13 in France). But I still have 617 errors.

I also reproduced a bug on http://refine.codefork.com/reconcile/viafproxy/BNF I had on my local version of refine_viaf-1.4.jar but I will make a new issue for this one (the Person type only is displayed in the reconciliation choices).
Thanks,

**Comment by JBPressac** [comment]
I noticed some problems after the reconciliation launched on http://refine.codefork.com/reconcile/viafproxy/BNF at 7:00 UTC (9:00 in France). The reconcialtion succesfully matched some individuals but did not get the BnF Id (FRBNF). For instance, the query "Jean-Guillaume Henry 1803 1880" succesfully matched with https://viaf.org/viaf/sourceID/BNF%7C10742576. But "Jean-François Alexandre 1804 1874 ", "Augustin Chatton 1825 1914 " and many others matched with https://viaf.org/viaf/sourceID/BNF%7C0.

![2016-06-17 11_20_53-prelib_personne140616 csv _ viaf - openrefine](https://cloud.githubusercontent.com/assets/6371750/16146228/49964488-347d-11e6-984e-ed85490bd773.png)

![2016-06-17 10_48_43-prelib_personne140616 csv _ viaf - openrefine](https://cloud.githubusercontent.com/assets/6371750/16145423/c89805f0-3478-11e6-8561-714341a96d53.png)

**Comment by codeforkjeff** [comment]
I've released v1.4.3 to fix the problem with the IDs for Alexandre and Chatton.

The "errors" in OpenRefine when you facet by score might be caused by a row having 0 candidates. If you no longer see errors in the logs with v1.4.3, then the 617 errors you're still seeing could be "normal", because VIAF did not match them to anything, so there is no score to use for the facet.

**Comment by JBPressac** [comment]
Thank you for the update !
